fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -47,6 +47,50 @@ pub fn from_url_to_url_auth(url: &Url) -> Result<(String, Auth)> {
         }),
         if url.username().is_empty() {
             Auth::None
+        } else if url.password().is_none() {
+            if let Some(cookie_file) = rpc_cookie_file {
+                let cookie = std::fs::read_to_string(cookie_file)?;
```

Comment:
Reading&parsing the cookie file is a natural boundary to make a separate functions (parsing from `&str` should be separate), that can be tested easily in isolation.

Reply:
Ok, I will isolate the functions.

Reply:
@VeronicaTee AFAIK parsing wouldn't be necessary, you could use the [`bitcoincore_rpc::Auth`](https://docs.rs/bitcoincore-rpc/0.16.0/bitcoincore_rpc/enum.Auth.html) with the [`CookieFile`](https://docs.rs/bitcoincore-rpc/0.16.0/bitcoincore_rpc/enum.Auth.html#variant.CookieFile) variant, and then get the authentication required by it's implemented methods.

Reply:
@oleonardolima Like the issue here https://github.com/fedimint/fedimint/issues/1994 you mean? OK.

Reply:
@VeronicaTee I mean, getting the given directory path from environment var would still be needed, but you can pass it as a [`std::path::PathBuf`](https://doc.rust-lang.org/nightly/std/path/struct.PathBuf.html) to `bitcoincore_rpc::Auth`.

So it wouldn't need to handle and parse it as it's below 😁 

https://github.com/fedimint/fedimint/blob/787b2a022a05845ed9913de0eadd689cd6e63a08/fedimint-bitcoind/src/bitcoincore_rpc.rs#L53-L83

Reply:
Ok, I understand you now. Thanks.